doc/readme.md

```diff
@@ -0,0 +1,67 @@
+
+Tests
+---
+
+Mathematics includes thousands of Unit tests to cover methods and operator overloads for the vector, matrix and quaterion types that can be found in the **Unity.Mathematics.Tests** project.
+The tests aim to exercise the entire set of Mathematics features using a combination of generated and manual tests. As Mathematics is intended to be used with the Burst compiler, we also include most of these tests as part of the Burst compiler tests in the **Burst.Compiler.IL.Tests** project. All shared tests between the two are in the `Tests/Shared` folder. A few tests have had to be omitted as the testing code it self uses unburstable code.
+
+Floating point precision and floating point overflow/underflow
+--------------------------------------------------------------
+
+Floating point precision in C# guarantees a minimum, but not a maximum precision. float might be evaluated as double (which mono likes to do) or double might be evaluated using extended precision (80bits). The precision is not even guaranteed to be consistent on a specific platform or even a concrete program. It might vary depending on how the JIT decides to optimize a particular piece of code!
+Anecdotely it seems these two otherwise semantically identical pieces of code can behave differently because of JIT optimization decisions (x86):
+1. Constants in array
+```C#
+var a = new double[2] = { -2.5e160, 1.2e160 };
+double d = System.Math.Sqrt(a[0] * a[0] + a[1] * a[1]);
+System.Console.WriteLine("" + d); // Debug Build: Infinity, Release Build: 2.86356e160.
+```
+2. Constants in local variables
+```C#
+double a = -2.6e160;
+double b = -1.2e160;
+double d = System.Math.Sqrt(a * a + b * b);
+System.Console.WriteLine("" + d); // Debug Build: Infinity, Release Build: Infinity.
+```
+You would expect the result to always be Infinity as the intermediate value inside the Sqrt (8.2e320) is outside the valid double precision range. But depending on the platform, whether optimizations are enabled and surprisingly even whether or not the constants come from a local variable or array, this might produce a finite value. Seemingly this happens because it might decide to perform the calculation in extended precision.
+This behavior is specified in the C# specification (9.3.7) where it states: "Floating-point operations may be performed with higher precision than the result type of the operation.".
+Due to this inconsistency, we have to avoid floating point overflow behavior, as it is simple not guaranteed even using microsofts own runtime!
```

Comment:
```suggestion
Due to this inconsistency, we have to avoid floating point overflow behavior, as it is simple not guaranteed even using Microsoft's own runtime!
```